### ISSUE TITLE
Random Crashes

### ISSUE THREAD

**Issue by stuntguy3000** [doitandbedone/ispyagentdvr-docker#42]
Hi,

This has been working fine for some time but suddenly it's been crashing a lot for no reason.
![image](https://user-images.githubusercontent.com/1522389/90389032-f7479a00-e0c7-11ea-8ce5-fe1b22fd8e3d.png)

What can be done to diagnose this?

Thanks

**Comment by ispysoftware** [comment]
that's coming from the object tracking code - you can try running ldd
libcvextern.so in the Agent directory to see what is missing - in the
meantime change the detector to use Simple to prevent crashes

**Comment by stuntguy3000** [comment]
Thanks for that suggestion, how do you do that within the Docker image?

**Comment by stuntguy3000** [comment]
root@NAS:/agent# ldd libcvextern.so
        linux-vdso.so.1 (0x00007ffe85265000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fce2eade000)
        libjpeg.so.8 => /usr/lib/x86_64-linux-gnu/libjpeg.so.8 (0x00007fce2e876000)
        libtbb.so.2 => /usr/lib/x86_64-linux-gnu/libtbb.so.2 (0x00007fce2e63a000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fce2e436000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fce2e217000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fce2e00f000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fce2dc86000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fce2d8e8000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fce2d6d0000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fce2d2df000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fce30452000)
root@NAS:/agent#

**Comment by ispysoftware** [comment]
I'll take a look at it tomorrow. 10pm here now

**Comment by stuntguy3000** [comment]
No worries, appreciated.

**Comment by ispysoftware** [comment]
https://github.com/emgucv/emgucv/issues/314

**Comment by stuntguy3000** [comment]
@ispysoftware As requested, I created the folder (on the docker instance) in tmp but still experience the same crash as soon as anything moves in the camera view.

```
dotnet: symbol lookup error: /agent/libcvextern.so: undefined symbol: _ZN2cv10TrackerKCF6ParamsC1Ev
```

**Comment by ispysoftware** [comment]
what version of agent are you using?

**Comment by stuntguy3000** [comment]
2.8.9, but I did try upgrading to 2.9 when it prompted which did not fix the issue.

**Comment by doitandbedone** [comment]
PR was queued. Image should update in a few minutes for master.

**Comment by ispysoftware** [comment]
i think the temp path varies according to which account it is running under - you could try running it under sudo dotnet agent.dll instead.
i managed to crash the OSX version with the tracker and fix it by creating the temp path.

**Comment by stuntguy3000** [comment]
Thanks @doitandbedone, I can confirm the crash still happens on 2.9 (just tested)

@ispysoftware how would I do this since it's a docker containerr?

**Comment by ispysoftware** [comment]
ok i repro'd it here - no idea what's causing it as the same code is working fine on osx and windows. I'll experiment with some alternative trackers.

**Comment by stuntguy3000** [comment]
Thanks.

**Comment by ispysoftware** [comment]
ok it's definitely some missing dependency in the linux opencv build. Unfortunately there's about 200 dependencies and no documentation on which relies on which so it might take a while to track it down.

**Comment by stuntguy3000** [comment]
Yikes!

Well, I certainly appreciate your efforts and would love to see this working.

Cheers

**Comment by ispysoftware** [comment]
uploading an update now

**Comment by stuntguy3000** [comment]
Thanks for this @ispysoftware, but after a full docker reinstall, I can confirm the bug still exists and it still crashes on motion.

**Comment by ispysoftware** [comment]
yeah i built it on the wrong version of ubuntu :( working on an update

**Comment by doitandbedone** [comment]
Oh nvm then ok. Also there's some sort of issue on the action and got stuck
on a loop.

**Comment by doitandbedone** [comment]
Wrong email thread... Please ignore

**Comment by ispysoftware** [comment]
you can try copying this to the docker install and see if it works - should do, working here. If it does i'll get it updated tomorrow.
(docker cp libcvextern.so agentdvr:/agent/libcvextern.so)
you'll need to restart the container to load it in

[
[libcvextern.zip](https://github.com/doitandbedone/ispyagentdvr-docker/files/5096259/libcvextern.zip)
](url)

**Comment by stuntguy3000** [comment]
Winner winner chicken dinner, tracking appears to be working! Testing functionality now...

**Comment by stuntguy3000** [comment]
New issue, tracking works but when using the Object or ONIFV mode, going into the settings causes this error:
![image](https://user-images.githubusercontent.com/1522389/90634748-57c10d80-e267-11ea-9ad6-63f246dcc656.png)
![image](https://user-images.githubusercontent.com/1522389/90634773-65769300-e267-11ea-99bd-e808961d0fc4.png)

**Comment by ispysoftware** [comment]
ok will check that out tomorrow

**Comment by ispysoftware** [comment]
it's been updated - fixed an issue with the object detector and a few other things - not sure what was causing that error you posted above - typically that would be an out of memory issue with opencv. Possibly fixed by the latest changes.
